application/src/main/java/org/thingsboard/server/service/script/RemoteJsInvokeService.java

```diff
@@ -192,12 +192,26 @@ protected ListenableFuture<Object> doInvokeFunction(UUID scriptId, String functi
         Futures.addCallback(future, new FutureCallback<TbProtoQueueMsg<JsInvokeProtos.RemoteJsResponse>>() {
             @Override
             public void onSuccess(@Nullable TbProtoQueueMsg<JsInvokeProtos.RemoteJsResponse> result) {
-                queueInvokeMsgs.incrementAndGet();
+                if (result != null) {
+                    JsInvokeProtos.JsInvokeResponse invokeResponse = result.getValue().getInvokeResponse();
+                    if (invokeResponse.getSuccess()) {
+                        queueInvokeMsgs.incrementAndGet();
+                    } else {
+                        JsInvokeProtos.JsInvokeErrorCode errorCode = invokeResponse.getErrorCode();
+                        final RuntimeException e = new RuntimeException(invokeResponse.getErrorDetails());
+                        if (JsInvokeProtos.JsInvokeErrorCode.TIMEOUT_ERROR.equals(errorCode)) {
+                            onScriptExecutionError(scriptId, e, scriptBody);
+                            queueTimeoutMsgs.incrementAndGet();
+                        } else if (JsInvokeProtos.JsInvokeErrorCode.COMPILATION_ERROR.equals(errorCode)) {
```

Comment:
What about JsInvokeErrorCode.RUNTIME_ERROR? When does this happen? Is it possible that the RUNTIME_ERROR will happen due to an error in the script (e.g. div by zero)?